Source/interactivity_manager_impl.cpp

```diff
@@ -1167,6 +1187,10 @@ interactivity_manager_impl::on_socket_message_received(
     _In_ const string_t& message
 )
 {
+    // We always return all messages from the service to to the developer, even the ones the SDK will process them.
+    std::shared_ptr<interactive_message_event_args> args = std::shared_ptr<interactive_message_event_args>(new interactive_message_event_args(message));
+    queue_interactive_event_for_client(L"", std::error_code(0, std::generic_category()), interactive_event_type::unknown, args);

```

Comment:
I get it does seem off, but (1) I don't want to break compat and (2) I do think there's legit reasons for sending messages as objects. Objects are easier for the developer to deal with. This generic message API was intended for advanced scenarios that we haven't wrapped yet in the SDK, but don't want to block devs on or functionality we'll never wrap. So I do think there's good reason to keep returning the object types.

---
In reply to: [132343459](https://github.com/mixer/interactive-cpp/pull/25#discussion_r132343459) [](ancestors = 132343459,132343213)